test/integration/controllermanager/seed/extensionscheck/extensionscheck_suite_test.go

```diff
@@ -89,6 +97,16 @@ var _ = BeforeSuite(func() {
 	mgr, err := manager.New(restConfig, manager.Options{
 		Scheme:             kubernetes.GardenScheme,
 		MetricsBindAddress: "0",
+		NewCache: cache.BuilderWithOptions(cache.Options{
+			SelectorsByObject: map[client.Object]cache.ObjectSelector{
```

Comment:
Using the label selector trick doesn't deflake the test by itself.
It allows running multiple tests in parallel against the same cluster, but it doesn't deflake the test.

> this allows running a test in parallel against the same existing cluster for deflaking and stress testing, even if it works with cluster-scoped resources that are visible to all parallel test runs

BTW: this is also why you can observe such a high failure rate before this PR – running multiple instances of the tested controller for all resources will likely cause failures. So I don't think the failures you shared in https://github.com/gardener/gardener/pull/6619#issuecomment-1235676149 are actually caused by the flake that you are trying to resolve.

Reply:
Yep, I thought so as well but since there are a couple of integration tests that use `ControllerInstallations` (that could run in parallel I guess) this should help in those cases as well.

Reply:
Hmmm, every envtest suite uses a dedicated test environment, so there will not be any conflicts between test suites 🤔

Reply:
Ahh right right, I guess I need more coffee 😅